extensions/amp-fx-collection/0.1/providers/amp-fx-presets.js

```diff
@@ -109,4 +109,70 @@ export const Presets = {
       });
     },
   },
+  'fade-in-scroll': {
+    isFxTypeSupported(win) {
+      user().assert(isExperimentOn(win, 'amp-fx-fade-in-scroll'),
+          'amp-fx-fade-in-scroll experiment is not turned on.');
+    },
+    userAsserts(element) {
+      if (!element.hasAttribute('data-margin-start') &&
+        !element.hasAttribute('data-margin-end')) {
+        return;
+      }
+      const marginStart = element.getAttribute('data-margin-start');
+      user().assert(parseFloat(marginStart) >= 0 &&
+        parseFloat(marginStart) < 100, 'data-margin-start must be a number ' +
+          'and be between 0% and 100% for: %s', element);
+      const marginEnd = element.getAttribute('data-margin-end');
+      user().assert(parseFloat(marginEnd) >= 0 &&
+        parseFloat(marginEnd) < 100, 'data-margin-start must be a number ' +
+          'and be between 0% and 100% for: %s', element);
+
+      user().assert(parseFloat(marginEnd) > parseFloat(marginStart),
+          'data-margin-end must be greater than data-margin-start for: %s',
+          element);
+    },
+    update(entry) {
+      const fxElement = this;
+      dev().assert(fxElement.adjustedViewportHeight_);
```

Comment:
nit: positionRect gets used throughout this method, maybe you can set an intermediate variable via destructuring.